ofrak_core/ofrak/core/ubi.py

```diff
@@ -0,0 +1,271 @@
+import subprocess
+import tempfile
+from dataclasses import dataclass
+import logging
+from typing import List
+import os
+
+from ofrak import Identifier, Analyzer
+from ofrak.component.packer import Packer
+from ofrak.component.unpacker import Unpacker
+from ofrak.model.component_model import ComponentExternalTool
+from ofrak.resource import Resource
+from ofrak.core.filesystem import File
+from ofrak.core.binary import GenericBinary
+from ofrak.resource_view import ResourceView
+
+from ofrak_type.range import Range
+
+from ubireader import ubi_io
+from ubireader.ubi import ubi as ubireader_ubi
+from ubireader.ubi.defines import UBI_EC_HDR_MAGIC, UBI_VID_HDR_MAGIC, PRINT_VOL_TYPE_LIST
+from ubireader.utils import guess_peb_size
+
+from ofrak.model.resource_model import index
+
+LOGGER = logging.getLogger(__name__)
+
+UBINIZE_TOOL = ComponentExternalTool(
+    "ubinize",
+    "http://www.linux-mtd.infradead.org/faq/ubi.html#L_ubi_mkimg",
+    install_check_arg="--help",
+    apt_package="mtd-utils",
+    brew_package="",  # This isn't compatible with macos, but there may be an alternative tool to do the bidding.
+)
+
+
+@dataclass
+class UbiVolume(ResourceView):
+    """
+    An UbiVolume is a volume entry in UBI. It typically contains an image of arbitrary data, typically a filesystem or a
+    log. They can be empty while reserving PEBs for future volume modification or for padding.
+
+    Volume information reflected in the 'config.ini' UBI volume entries expected by `ubinize` are stored here. Also see:
+    http://www.linux-mtd.infradead.org/faq/ubi.html#L_ubi_mkimg and
+
+    :cvar id: The assigned volume ID within the UBI image
```

Comment:
Have you tested that `cvar` renders correctly in the built docs? I know that `var` and `ivar` both work

I think technically these are not class variables -- it looks like class variables are specifically skipped by dataclass: https://docs.python.org/3/library/dataclasses.html#class-variables. 

Let's just change all of these to `:var ...`?